### PR TITLE
docs(samples): explicitly add bq to samples reqs, upgrade grpc to fix bug on m1

### PR DESCRIPTION
**Pull request by leahecole**
Related to #1260 and #1262

Can confirm these linked issues definitely happen with M1, but is fixed if you specifically add grpc as an explicit dependency

### CODE REVIEW

**Review by loferris — CHANGES_REQUESTED** [review]
Hi @leahecole, looking at the system test failures, I think we can fix this if we update the required versions in `setup.py` (project root) and the constraint files for at least 3.8 and 3.10 in the `testing` folder (project root).

**Review by parthea — APPROVED** [review]

**Comment by parthea** [comment]
Should we also update the grpc requirements in setup.py ? https://github.com/googleapis/python-bigquery/blob/main/setup.py#L32

**Comment by leahecole** [comment]
> Hi @leahecole, looking at the system test failures, I think we can fix this if we update the required versions in `setup.py` (project root) and the constraint files for at least 3.8 and 3.10 in the `testing` folder (project root).

To clarify, the constraint being the grpc version, not the BQ version we also added, right, @loferris?

**Review by loferris — CHANGES_REQUESTED** [review]
Looking at the Kokoro failures, I'm seeing a flag to update the 3.7 constraints file (in the `testing` folder) to have the correct `grpcio` version. 
Screen Shot 2022-07-18 at 1 33 21 PM
.

**Comment by loferris** [comment]
> > Hi @leahecole, looking at the system test failures, I think we can fix this if we update the required versions in `setup.py` (project root) and the constraint files for at least 3.8 and 3.10 in the `testing` folder (project root).
> 
> To clarify, the constraint being the grpc version, not the BQ version we also added, right, @loferris?

From what I can tell from the Kokoro test failures, that seems right to me!!

**Comment by parthea** [comment]
`testing/constraints-3.6.txt` could be deleted since we're no longer supporting python 3.6

**Comment by leahecole** [comment]
Ok! I'll make those changes. Thanks for the explanation - I've never touched these constraints files before so I wanted to be super sure I was doing the right thing 😁

File: testing/constraints-3.10.txt [hunk 1]
```diff
@@ -0,0 +1 @@
+grpcio==1.47.0
```
[the remark this answers is not in the document]
Comment (loferris) [reply testing/constraints-3.10.txt:1]: I think we had to add this explicitly to get the system tests (if not, would be happy to delete!!), but I'd suggest I make a follow up issue to clean up the versioning across different versions.

**Review by loferris — APPROVED** [review]
I'm going to approve and merge, but I do want to follow up with @parthea about versioning overall.

**Comment by leahecole** [comment]
SGTM - let me know what you end up deciding!

**Pull request merged**